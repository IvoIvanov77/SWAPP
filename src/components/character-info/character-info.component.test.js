import React from 'react';
import { shallow } from 'enzyme';
import CharacterInfo from './character-info.component';

describe('CharacterInfo component', () => {
  let wrapper;
  const name = 'name';
  const image = 'image';
  const height = '100';
  const homeworld = { name: 'homeworld' };
  const species = { name: 'species' };

  beforeEach(() => {
    const person = {
      name,
      image,
      height,
      homeworld,
      species,
    };
    wrapper = shallow(<CharacterInfo person={person} />);
  });

  it('should render CharacterInfo component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should pass image to CharacterImage as the prop src', () => {
    expect(wrapper.find('CharacterImage').prop('src')).toBe(image);
  });

  it('should element contains person fields', () => {
    expect(wrapper.find('CharacterName').text()).toBe(name);
    expect(
      wrapper
        .find('span')
        .at(0)
        .text(),
    ).toBe(height);
    expect(
      wrapper
        .find('span')
        .at(1)
        .text(),
    ).toBe(species.name);
    expect(
      wrapper
        .find('span')
        .at(2)
        .text(),
    ).toBe(homeworld.name);
  });
});
