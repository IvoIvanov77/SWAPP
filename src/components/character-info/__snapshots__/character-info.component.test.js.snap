// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterInfo component should render CharacterInfo component 1`] = `ShallowWrapper {}`;
